File: api/src/test/java/net/blueshell/api/factory/dto/RedirectDTOFactory.java
package net.blueshell.api.factory.dto;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.dto.RedirectDTO;
import org.springframework.stereotype.Component;

/**
 * Factory for RedirectDTO test instances.
 */
@Component
@RequiredArgsConstructor
public class RedirectDTOFactory extends BaseDtoFactory<RedirectDTO> {

    private final TelemetryDTOFactory telemetryFactory;

    @Override
    public Class<RedirectDTO> targetType() {
        return RedirectDTO.class;
    }

    @Override
    public RedirectDTO createBasic() {
        RedirectDTO dto = new RedirectDTO();
        dto.setCreatedAt(now());
        dto.setTelemetry(telemetryFactory.createBasic());
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/RedirectDTOFactory.kt
package net.blueshell.api.factory.dto

import net.blueshell.api.telemetry.web.dto.RedirectDTO
import org.springframework.stereotype.Component

/**
 * Factory for RedirectDTO test instances.
 */
@Component
class RedirectDTOFactory(
    private val telemetryFactory: TelemetryDTOFactory
) : BaseDtoFactory<RedirectDTO>() {

    override fun targetType(): Class<RedirectDTO> = RedirectDTO::class.java

    override fun createBasic(): RedirectDTO {
        val dto = RedirectDTO()
        dto.createdAt = now()
        dto.telemetry = telemetryFactory.createBasic()
        return dto
    }
}
